src/aiidalab_qe/plugins/xps/result.py

```diff
@@ -0,0 +1,247 @@
+        )
+        fill = ipw.Checkbox(
+            description="Fill",
+            value=True,
+            disabled=False,
+            style={"description_width": "initial"},
+        )
+        paras = ipw.HBox(
+            children=[
+                gamma,
+                sigma,
+            ]
+        )
+        # get data
+        (
+            chemical_shifts,
+            binding_energies,
+            equivalent_sites_data,
+        ) = export_xps_data(self.outputs.xps)
+        self.spectrum_select_options = [
+            key.split("_")[0] for key in chemical_shifts.keys()
+        ]
+        spectrum_select = ipw.Dropdown(
+            description="",
+            disabled=False,
+            value=self.spectrum_select_options[0],
+            options=self.spectrum_select_options,
+            layout=ipw.Layout(width="20%"),
+        )
+        # init figure
+        g = go.FigureWidget(
+            layout=go.Layout(
+                title=dict(text="XPS"),
+                barmode="overlay",
+            )
+        )
+        g.layout.xaxis.title = "Chemical shift (eV)"
+        g.layout.xaxis.autorange = "reversed"
+        #
+        spectra = xps_spectra_broadening(
+            chemical_shifts, equivalent_sites_data, gamma=gamma.value, sigma=sigma.value
+        )
+        # only plot the selected spectrum
+        for site, d in spectra[spectrum_select.value].items():
+            g.add_scatter(x=d[0], y=d[1], fill="tozeroy", name=site.replace("_", " "))
+
+        def response(change):
+            data = []
+            if spectra_type.value == "chemical_shift":
+                points = chemical_shifts
+                xaxis = "Chemical Shift (eV)"
+            else:
+                points = binding_energies
+                xaxis = "Binding Energy (eV)"
+            #
+            spectra = xps_spectra_broadening(
+                points, equivalent_sites_data, gamma=gamma.value, sigma=sigma.value
+            )
+
+            for site, d in spectra[spectrum_select.value].items():
+                data.append(
+                    {
+                        "x": d[0],
+                        "y": d[1],
+                        "site": site,
+                    }
+                )
+
+            with g.batch_update():
+                if len(g.data) == len(data):
+                    for i in range(len(data)):
+                        g.data[i].x = data[i]["x"]
+                        g.data[i].y = data[i]["y"]
+                        if fill.value:
+                            g.data[i].fill = "tozeroy"
+                        else:
+                            g.data[i].fill = None
+                        g.data[i].name = data[i]["site"].replace("_", " ")
+
+                else:
+                    g.data = []
+                    for d in data:
+                        g.add_scatter(
+                            x=d["x"], y=d["y"], fill=fill.value, name=d["site"]
+                        )
```

Comment:
This section here causes a crash because you're calling the value from the UI element rather than converting it to a type of fill method (e.g. "tozeroy"):
```suggestion
                        if fill.value:
                            fill_type = "tozeroy"
                        else:
                            fill_type = None
                        g.add_scatter(
                            x=d["x"], y=d["y"], fill=fill_type, name=d["site"]
                        )
```

Reply:
thanks! I changed it.